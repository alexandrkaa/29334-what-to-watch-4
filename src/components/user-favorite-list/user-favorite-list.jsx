import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {moviesListType} from '../../types/types.js';
import {connect} from 'react-redux';
import {AppRoutes} from '../../consts/consts.js';
import {Redirect} from 'react-router-dom';
import {getMoviesDataFromUserFavoriteList, hasUserLogined} from '../../reducer/selectors.js';
import Header from '../header/header.jsx';
import Footer from '../footer/footer.jsx';
import MoviesList from '../movie-list/movie-list.jsx';
import UserProfile from '../user-profile/user-profile.jsx';

class UserFavoriteList extends PureComponent {

  render() {
    const {moviesList, isAuthorized} = this.props;
    if (!isAuthorized) {
      return (
        <Redirect to={AppRoutes.LOGIN_PAGE} />
      );
    }

    return (
      <div className="user-page">
        <Header headerClassName={`user-page__head`}>
          <h1 className="page-title user-page__title">My list</h1>
          <UserProfile />
        </Header>
        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>
          <div className="catalog__movies-list">
            <MoviesList
              moviesList={moviesList}
            />
          </div>
        </section>
        <Footer />
      </div>
    );
  }
}

UserFavoriteList.propTypes = {
  moviesList: moviesListType,
  isAuthorized: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => {
  return {
    moviesList: getMoviesDataFromUserFavoriteList(state),
    isAuthorized: hasUserLogined(state),
  };
};

export {UserFavoriteList};
export default connect(mapStateToProps)(UserFavoriteList);
