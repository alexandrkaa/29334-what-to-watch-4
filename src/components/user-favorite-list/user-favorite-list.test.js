import React from 'react';
import {MyList} from './user-favorite-list.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ShallowRenderer();

const mockData = {
  moviesList: [
    {
      id: 4,
      image: `img/we-need-to-talk-about-kevin.jpg`,
      movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
      movieDate: `2000`,
      movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
      movieDirector: `Wes Anderson`,
      movieGenre: `Thriller`,
      movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
      movieRatingCount: `349 ratings`,
      movieRatingLevel: `Bad`,
      movieRatingScore: `3`,
      movieStarring: [`Jude Law`, `Willem Dafoe`],
      title: `Dardjeeling Limited`,
      moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      backgroundColor: `#000000`,
      movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      isFavorite: false,
    },
    {
      id: 3,
      image: `img/aviator.jpg`,
      movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
      movieDate: `2008`,
      movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
      movieDirector: `Christopher Nolan`,
      movieGenre: `Kids & Family`,
      movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
      movieRatingCount: `272 ratings`,
      movieRatingLevel: `Normal`,
      movieRatingScore: `2`,
      movieStarring: [`Jude Law`, `Willem Dafoe`],
      title: `Johnny English`,
      moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      backgroundColor: `#000000`,
      movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      isFavorite: false,
    }
  ],
};

it(`<MyList /> should match snapshot without error`, () => {
  const checkAuth = jest.fn();
  renderer.render(<MyList moviesList={mockData.moviesList} isAuthorized={true} checkAuth={checkAuth} />);
  const result = renderer.getRenderOutput();
  expect(result.type).toBe(`div`);
  expect(result).toMatchSnapshot();
});
