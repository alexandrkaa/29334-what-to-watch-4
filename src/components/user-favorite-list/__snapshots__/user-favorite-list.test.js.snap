// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserFavoriteList /> should match snapshot without error 1`] = `
<div
  className="user-page"
>
  <Header
    headerClassName="user-page__head"
  >
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <Memo(Connect(UserProfile)) />
  </Header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <Memo(MoviesList)
        moviesList={
          Array [
            Object {
              "backgroundColor": "#000000",
              "id": 4,
              "image": "img/we-need-to-talk-about-kevin.jpg",
              "isFavorite": false,
              "movieBackground": "img/bg-the-grand-budapest-hotel.jpg",
              "movieDate": "2000",
              "movieDescription": "In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.",
              "movieDirector": "Wes Anderson",
              "movieGenre": "Thriller",
              "movieImage": "img/the-grand-budapest-hotel-poster.jpg",
              "moviePreview": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "movieRatingCount": "349 ratings",
              "movieRatingLevel": "Bad",
              "movieRatingScore": "3",
              "movieStarring": Array [
                "Jude Law",
                "Willem Dafoe",
              ],
              "movieVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "title": "Dardjeeling Limited",
            },
            Object {
              "backgroundColor": "#000000",
              "id": 3,
              "image": "img/aviator.jpg",
              "isFavorite": false,
              "movieBackground": "img/bg-the-grand-budapest-hotel.jpg",
              "movieDate": "2008",
              "movieDescription": "In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.",
              "movieDirector": "Christopher Nolan",
              "movieGenre": "Kids & Family",
              "movieImage": "img/the-grand-budapest-hotel-poster.jpg",
              "moviePreview": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "movieRatingCount": "272 ratings",
              "movieRatingLevel": "Normal",
              "movieRatingScore": "2",
              "movieStarring": Array [
                "Jude Law",
                "Willem Dafoe",
              ],
              "movieVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "title": "Johnny English",
            },
          ]
        }
      />
    </div>
  </section>
  <Footer />
</div>
`;
